### PR TITLE
feat: Add scopes to tokens issued by JwtService (DEV-3451)

### PR DESCRIPTION
**Pull request by seakayone**
- **Move code to KnoraUserToUserConverter**
- **Migrate JwtServiceLiveSpec to zio-test**
- **Calculate scope values based on the user's permissions and add them to the access token as scope in jwt**

### Task Description/Number

Issue Number: DEV-

### PR Type

- [ ] build/chore: maintenance tasks (no production code change)
- [ ] docs: documentation changes (no production code change)
- [x] feat: represents new features
- [ ] fix: represents bug fixes
- [ ] perf: performance improvements
- [ ] refactor: represents production code refactoring
- [ ] test: adding or refactoring tests (no production code change)
- [ ] deprecated: Deprecation warning (ideally referencing a migration guide)

### Basic requirements for bug fixes and features

- [ ] Tests for the changes have been added
- [ ] Docs have been added / updated

### Does this PR introduce a breaking change?

- [ ] Yes

### Does this PR change client-test-data?

- [ ] Yes

### CODE REVIEW

File: webapi/src/main/scala/org/knora/webapi/slice/infrastructure/Scope.scala [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+  def merge(one: ScopeValue, two: ScopeValue): Set[ScopeValue] =
+    (one, two) match {
+      case (Admin, _) | (_, Admin)            => Set(Admin)
+      case (Write(p1), Write(p2)) if p1 == p2 => Set(Write(p1))
+      case (Write(p1), Read(p2)) if p1 == p2  => Set(Write(p1))
+      case (Read(p1), Write(p2)) if p1 == p2  => Set(Write(p1))
+      case (Read(p1), Read(p2)) if p1 == p2   => Set(Read(p1))
+      case (a, b)                             => Set(a, b)
+    }
```
Comment (siers) [inline webapi/src/main/scala/org/knora/webapi/slice/infrastructure/Scope.scala:59]: This finds the "biggest" items of Set(one, two) by partial ordering Admin > Write > Read, with Write/Reads being not comparable if their contents differ, but there's not stdlib support, I'm afraid.

Comment (siers) [reply webapi/src/main/scala/org/knora/webapi/slice/infrastructure/Scope.scala:59]: Reason why I'm thinking about this is, if you would phrase it this way, then you wouldn't have the have the second filtering in `def +(addThis)`... Nevertheless, it looks to me like you don't need it anyway, let me consult your tests.

Comment (siers) [reply webapi/src/main/scala/org/knora/webapi/slice/infrastructure/Scope.scala:59]: fix pushed

**Review by mpro7 — APPROVED** [review]

File: webapi/src/main/scala/org/knora/webapi/slice/infrastructure/Scope.scala (hunk 1 above)
Comment (seakayone) [reply webapi/src/main/scala/org/knora/webapi/slice/infrastructure/Scope.scala:48]: Thanks

**Pull request merged**